sqlglot/optimizer/simplify.py

```diff
@@ -37,10 +37,10 @@ def _simplify(expression, root=True):
         exp.replace_children(node, lambda e: _simplify(e, False))
         node = simplify_not(node)
         node = flatten(node)
-        node = simplify_connectors(node)
+        node = simplify_connectors(node, root)
```

Comment:
seems like root is missing in a couple of the other functions, let's add root for every function that relies on same_parent

Reply:
Done. But it bled into normalize.